### ISSUE TITLE
ipa file generated is huge

### ISSUE THREAD

**Issue by ahmed-wagdi** [expo/expo#205]
After running `exp build:ios` the ipa file generated was 150 MB is this normal and is there anyway to reduce it?

I'm using expo version 16.0.0 and exp version 41.0.0

**Comment by dikaiosune** [comment]
Hi! Please see [this section of the docs](https://docs.expo.io/versions/v17.0.0/guides/building-standalone-apps.html#4-wait-for-it-to-finish-building) and the related [App Thinning docs](https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/AppThinning/AppThinning.html) from Apple.

The remaining size of the bundle (around 20MB) contains the Expo runtime and makes sure that your users have all of the native code required to receive any updates you may publish.

**Comment by ahmed-wagdi** [comment]
@dikaiosune I've been looking at the app thinning docs but it's not entirely clear how it's done. From the looks of it it's done through Xcode but i can't seem to open the ipa file through Xcode.

**Comment by satya164** [comment]
> The store and operating system optimize the installation of iOS, tvOS, and watchOS apps by tailoring app delivery to the capabilities of the user’s particular device, with minimal footprint. 

>  The store will create and deliver different variants based on the devices your app supports. Image resources are sliced according to their resolution and device family. GPU resources are sliced according to device capabilities. For tvOS apps, assets in catalogs shared between iOS and tvOS targets are sliced and large app icons are removed.

TLDR; The app store does the app thinning.

**Comment by ahmed-wagdi** [comment]
@satya164 what if i'm not publishing through the app store? I'm building an in-house app for an organization so it's not going to be on the app store, it will be distributed OTA similar to this: https://stackoverflow.com/questions/26042508/over-the-air-ota-ios-ipa-file-distribution-for-public

**Comment by ahmed-wagdi** [comment]
any ideas?

**Comment by brentvatne** [comment]
we don't explicitly support building an app for enterprise distribution (afaik, although i imagine it works fine) or without using bitcode. you could follow the steps described here but disable bitcode in xcode before doing it: https://github.com/expo/expo#ios-1 -- i have no idea if this will "just work" though and it's not something we plan to support in the near future. you can create a feature request here: https://expo.canny.io/feature-requests

another option is to just get people to download the expo client, then you don't need to go through the hassle of building an ipa and paying for an enterprise distribution license.

**Comment by vincent-lu** [comment]
Hi @brentvatne I'm having the same trouble as @ahmed-wagdi. My app ends up around 140mb. Do you know what kind of file size I will be looking at if I go thru the process listed here https://github.com/expo/expo#ios-1 ? Alternatively, since our app is protected by user auth system anyway, would you recommend us just submit the app to Apple so our users can download them from App store instead? Thanks in advance for answering my question :)

@ahmed-wagdi: did you have any luck building `ipa` with bitcode disabled?

**Comment by ahmed-wagdi** [comment]
@vincent-lu I had to eject and build the ipa myself using xcode which was a huge headache. That being said i don't think you need to worry about the app size if you're going to submit it to the app store for the reasons satya164 stated above.

I had to go through it because my app won't be distributed through the app store, it uses enterprise distribution.

**Comment by vincent-lu** [comment]
@ahmed-wagdi going App Store is a potential path for us but all things being equal I'd prefer to be able to build `ipa` to be distributed with our enterprise license.

Did you manage to build a non-app-thinnned `ipa` with xcode successfully?

**Comment by ahmed-wagdi** [comment]
Building with xcode generated an IPA that ranged from 10-15mb

**Comment by vincent-lu** [comment]
Ok that's significant reduction in size. Seems to be worth doing. Thanks @ahmed-wagdi

**Comment by ShubhGar** [comment]
set bitcode enable to yes,which make your app in binary form and reduce the size also in iOS 9 or later,Apple slice the app(App thinning) according to device.

**Comment by terribleben** [comment]
Hey, just going to provide some information on this wide-ranging thread of theories:

- If you build with `exp build:ios` you get an ipa with bitcode enabled which, in raw ipa form, will be very large.
- If you submit that ipa to the app store, the actual app store listing will (at time of writing) be something on the order of 30mb.
- The file size is because your app contains the whole Expo SDK. It's on our roadmap to allow more flexibility around this in the future.
- `exp build` also supports enterprise builds (which was not true when this thread was originally written). You will need to provide your own credentials during the build process, rather than letting Expo manage them for you.
- Lastly, if 30mb is too large for your users, you can run `exp detach`, allowing you to [manage your own Xcode project](https://docs.expo.io/versions/latest/expokit/) however you want. You get full control over build settings. However, this is more work, and also means you can no longer use `exp build`.

**Comment by ShubhGar** [comment]
@johanarnor Enable Bit code

**Comment by johanarnor** [comment]
@ShubhGar I don't follow. How do I enable bit code without ejecting (i.e. without Xcode)?

**Comment by ShubhGar** [comment]
@johanarnor 
You can list all the keys and values of your project build settings with the command

xcodebuild -showBuildSettings
Then find one or more keys you want to override, To ENABLE_BITCODE. Use Command:

xcodebuild -target <your target> -configuration <your configuration> ENABLE_BITCODE=YES

**Comment by johanarnor** [comment]
@ShubhGar Thanks for the reply, but I don't think this is possible. Since I haven't detached I don't have a Xcode project and is limited to expo's tools.
